addons/l10n_se_ocr/models/account_journal.py

```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, api, _
+from odoo.exceptions import UserError
+
+
+class AccountJournal(models.Model):
+    _inherit = 'account.journal'
+
+    invoice_reference_model = fields.Selection(selection_add=[('se_ocr2', 'Sweden OCR Level 1 & 2'), ('se_ocr3', 'Sweden OCR Level 3'), ('se_ocr4', 'Sweden OCR Level 4')])
+    l10n_se_invoice_ocr_length = fields.Integer(string='OCR Number Length', help="Total length of OCR Reference Number including checksum.", default=6)
+
+    @api.onchange('l10n_se_invoice_ocr_length')
+    def _onchange_l10n_se_invoice_ocr_length(self):
```

Comment:
Check the latest changes. It should be fixed according to your requests.

Reply:
I saw you changed a lot of things.  I am talking about this onchange itself to change it to a constraint.  Like https://github.com/odoo/enterprise/blob/53aafc1fd089cc61210f848fd30cc36204aa9d2d/account_asset/models/account_asset.py#L710 or https://github.com/odoo/odoo/blob/fc885499a3c75d8d564569ec4cea41a8a64f3164/addons/mrp/models/mrp_bom.py#L87

Reply:
So basicly you want something like this:

```
    @api.constraint('l10n_se_invoice_ocr_length')
    def _check_l10n_se_invoice_ocr_length(self):
        if self.l10n_se_invoice_ocr_length < 5:
            return {'warning': {'title': _('Warning'), 'message': _('OCR Reference Number length need to be greater than 5. Please correct settings under invoice journal settings.')}}
```
Instead of onchange?

Do you want to have the warning as return value or as raise ValidationError call?

Reply:
If it is with the constraint, we need the raise.  (api.constrains)  But I did not see you had changed the raise to warning already.  If you are sure it does not give a traceback further on, then it can stay this way. @daniel-stenlov

Reply:
Ok. I have modified it as to constraint with raise...